docker/src/main/java/org/arquillian/cube/impl/util/BindingUtil.java

```diff
@@ -20,12 +20,21 @@ private BindingUtil() {
     }
 
     public static Binding binding(DockerClientExecutor executor, String cubeId) {
-        InspectContainerResponse inspectResponse = executor.getDockerClient().inspectContainerCmd(cubeId).exec();
+        InspectContainerResponse inspectResponse = executor.getDockerClient().inspectContainerCmd( cubeId ).exec();
 
         HostConfig hostConfig = inspectResponse.getHostConfig();
+
+        /**
+         * This isn't actually the gateway IP of the host, it's the ip of the gateway within the docker runtime,
+         * which may not be accessible to us.  We need the IP the client uses, that will be routable, and will
+         * allow us to validate the ports in later phases
         String gatewayIp = inspectResponse.getNetworkSettings().getGateway();
 
-        Binding binding = new Binding(gatewayIp);
+         **/
+
+        final String dockerIp = executor.getDockerUri().getHost();
+        Binding binding = new Binding(dockerIp);
```

Comment:
Understood so what we are using here is the ip of the docker server. client(ip1) - docker(ip2) - contianer(ip3) which in my schema is the ip2. So in ping await strategy we are going to ping ip2 and the port defined in port binding.  So if dockerserver is in localhost then it is ok, if it is remote ip then we need another wait strategy which executes an `ss` command and return when the port is available.  Is it correct or ping strategy will work in case of remote host (like boot2docker) as well?